src/plugins/NEMD/MettDeamon.cpp

```diff
@@ -1584,25 +1678,25 @@ void Reservoir::sortParticlesToBins()
 	cout << "["<<ownRank<<"]: _particleVector.size()=" << _particleVector.size() << endl;
 
 	cout << "["<<ownRank<<"]: bbMin ="
-			<< domainDecomp.getBoundingBoxMin(0, domain) << ", "
-			<< domainDecomp.getBoundingBoxMin(1, domain) << ", "
-			<< domainDecomp.getBoundingBoxMin(2, domain) << "; bbMax = "
-			<< domainDecomp.getBoundingBoxMax(0, domain) << ", "
-			<< domainDecomp.getBoundingBoxMax(1, domain) << ", "
-			<< domainDecomp.getBoundingBoxMax(2, domain) << endl;
+			<< domainDecomp->getBoundingBoxMin(0, domain) << ", "
+			<< domainDecomp->getBoundingBoxMin(1, domain) << ", "
+			<< domainDecomp->getBoundingBoxMin(2, domain) << "; bbMax = "
+			<< domainDecomp->getBoundingBoxMax(0, domain) << ", "
+			<< domainDecomp->getBoundingBoxMax(1, domain) << ", "
+			<< domainDecomp->getBoundingBoxMax(2, domain) << endl;
 #endif
 	std::vector< std::vector<Molecule> > binVector;
 	binVector.resize(numBins);
 	uint32_t nBinIndex;
-	for(auto&& mol:_particleVector)
+	for(auto mol:_particleVector)  // auto&&
 	{
 		// possibly change component IDs
 		this->changeComponentID(mol, mol.componentid() );
 		double y = mol.r(1);
 		nBinIndex = floor(y / _dBinWidth);
 //		cout << "["<<ownRank<<"]: y="<<y<<", nBinIndex="<<nBinIndex<<", _binVector.size()="<<binVector.size()<<endl;
 		mardyn_assert(nBinIndex < binVector.size() );
-		mol.setr(1, y - nBinIndex*_dBinWidth);  // positions in slabs related to origin (x,y,z) == (0,0,0)
+//		mol.setr(1, y - nBinIndex*_dBinWidth);  // positions in slabs related to origin (x,y,z) == (0,0,0)
```

Comment:
is this commented out on purpose? If so -> remove it

Reply:
Ok, removed it.